crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -432,28 +594,59 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_validate_script() {
+        let transaction = Transaction {
+            version: Version(1),
+            lock_time: LockTime::from_height(0).unwrap(),
+            input: vec![TxIn {
+                previous_output: OutPoint {
+                    txid: Txid::from_str(
+                        "10ddd830599b17cc690535f7df28a84466eaca3c22f0d55b79023b6570f4fbc5",
+                    )
+                    .unwrap(),
+                    vout: 0,
+                },
+                script_sig: ScriptBuf::from_hex("00473044022100d0ed946330182916da16a6149cd313a4b1a7b41591ee52fb3e79d64e36139d66021f6ccf173040ef24cb45c4db3e9c771c938a1ba2cf8d2404416f70886e360af401475121022afc20bf379bc96a2f4e9e63ffceb8652b2b6a097f63fbee6ecec2a49a48010e2103a767c7221e9f15f870f1ad9311f5ab937d79fcaeee15bb2c722bca515581b4c052ae")
+                    .unwrap(),
+                sequence: Sequence::MAX,
+                witness: Witness::default(),
+            }],
+            output: vec![TxOut {
+                value: Amount::from_sat(1),
+                script_pubkey: ScriptBuf::from_hex(
+                    "76a9146f34d3811aded1df870359f311c2a11a015e945388ac",
+                )
+                .unwrap(),
+            }],
+        };
+
+        let txout = TxOut {
+            value: Amount::from_sat(5_000_350_000),
+            script_pubkey: ScriptBuf::from_hex("a914748284390f9e263a4b766a75d0633c50426eb87587")
+                .unwrap(),
+        };
+
+        let utxo_data = UtxoData::new(txout, 0, 0);
+
+        let mut utxo_map = HashMap::new();
+
+        utxo_map.insert(transaction.input[0].previous_output, utxo_data.clone());
+
+        assert!(dbg!(Consensus::validate_script(&transaction, 0, &utxo_map)).is_ok());
+    }
+
     #[test]
     fn test_validate_get_out_value() {
         let output = TxOut {
             value: Amount::from_sat(5_000_350_000),
             script_pubkey: ScriptBuf::from_hex("41047eda6bd04fb27cab6e7c28c99b94977f073e912f25d1ff7165d9c95cd9bbe6da7e7ad7f2acb09e0ced91705f7616af53bee51a238b7dc527f2be0aa60469d140ac").unwrap(),
         };
         let mut value_var = 0;
-        assert!(Consensus::get_out_value(&output, &mut value_var).is_ok());
+        Consensus::get_out_value(&output, &mut value_var);
         assert_eq!(value_var, 5_000_350_000);
     }
 
-    #[test]
-    fn test_validate_script_size() {
-        //the case when the script is too big
-        let invalid_script = ScriptBuf::from_hex(&format!("{:0>1220}", "")).unwrap();
-        //the valid script < 520 bytes
-        let valid_script =
-            ScriptBuf::from_hex("76a9149206a30c09cc853bb03bd917a4f9f29b089c1bc788ac").unwrap();
-        assert!(Consensus::validate_script_size(&valid_script).is_ok());
-        assert!(Consensus::validate_script_size(&invalid_script).is_err());
-    }
-
     #[test]
     fn test_validate_coinbase() {
```

Comment:
Here, since `verify_coinbase` would have been modified (taking a reference of tx and not taking the index) we can just:

```rust
        let valid_one = coinbase(true);
        let invalid_one = coinbase(false);
        // The case that should be valid
        assert!(Consensus::verify_coinbase(&valid_one).is_ok());
        // Invalid coinbase script
        assert_eq!(
            Consensus::verify_coinbase(&invalid_one)
                .unwrap_err()
                .to_string(),
            "Invalid coinbase: \"Invalid ScriptSig size\""
        );
```